### PR TITLE
[DTSSTCI-578] Updated WA release scripts and templates

### PR DESCRIPTION
**Pull request by oleon23**
### Change description ###
This is the deployment PR for Work Allocation for Special Tribunals

### JIRA link ###
https://tools.hmcts.net/jira/browse/DTSSTCI-578

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)
- [ ] `enable-e2e-tests` label can be used to run the e2e tests before QA handover and before release (required)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket has been reviewed by QA
- [ ] the user story has been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: charts/sptribs-case-api/Chart.yaml [hunk 1]
```diff
@@ -43,3 +43,16 @@ dependencies:
     version: 1.0.31
     repository: 'https://hmctspublic.azurecr.io/helm/v1/repo/'
     condition: em-stitching.enabled
+#    Work Allocation
+  - name: wa
+    version: 1.0.5
```
Comment (tomxelliott) [inline charts/sptribs-case-api/Chart.yaml:48]: Can we make it:
    version: ~1.0.5

File: charts/sptribs-case-api/values.preview.template.yaml [hunk 2]
```diff
@@ -548,14 +573,35 @@ em-stitching:
       S2S_BASE_URI: http://rpe-service-auth-provider-aat.service.core-compute-aat.internal
       DM_STORE_APP_URL: http://dm-store-aat.service.core-compute-aat.internal
       CDAM_URL: http://${SERVICE_NAME}-ccd-case-document-am-api
-      SPRING_DATASOURCE_HOST: ${SERVICE_NAME}-postgresql
-      SPRING_DATASOURCE_NAME: emstitch
+      SPRING_DATASOURCE_HOST: '{{ tpl .Values.global.postgresHostname $}}'
+      SPRING_DATASOURCE_NAME: " {{ .Values.global.databaseNamePrefix }}emstitch"
```
Comment (hannah38) [inline charts/sptribs-case-api/values.preview.template.yaml:577]: There's just an extra space at the front of this one

File: build.gradle [hunk 3]
```diff
@@ -504,6 +504,10 @@ dependencies {
   implementation group: 'com.github.ben-manes.caffeine', name: 'caffeine', version: '3.1.8'
 
   implementation 'org.camunda.bpm:camunda-external-task-client:7.21.0'
+//  implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
```
Comment (rajaram-nagarajan) [inline build.gradle:507]: Shall we remove these commented lines?

Comment (oleon23) [reply build.gradle]: Removed - yep these are redundant now these were used for part of the attempt at local implementation/ with cftlib

**Review by rajaram-nagarajan — APPROVED** [review]

File: bin/wa/import-wa-bpmn-diagram.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+set -eu
+workspace=${1}
+
+serviceToken=$($(realpath $workspace)/utils/idam-lease-service-token.sh sptribs_case_api \
+  $(docker run --rm toolbelt/oathtool --totp -b ${S2S_SECRET:-AABBCCDDEEFFGGHH}))
+
+filepath="$(realpath $workspace)/resources"
+
+for file in $(find ${filepath} -name '*.bpmn')
```
Comment (gmmagruder) [inline bin/wa/import-wa-bpmn-diagram.sh:11]: Do we have .bpmn files and where do we get them from?

**Review by gmmagruder — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerReferToLegalOfficer.java [hunk 5]
```diff
@@ -40,29 +43,39 @@ public class CaseWorkerReferToLegalOfficer implements CCDConfig<CaseData, State,
...
+                .showEventNotes()
+                .aboutToStartCallback(this::aboutToStart)
+                .aboutToSubmitCallback(this::aboutToSubmit)
+                .submittedCallback(this::submitted)
+                .grant(CREATE_READ_UPDATE, SUPER_USER,
+                    ST_CIC_HEARING_CENTRE_ADMIN, ST_CIC_HEARING_CENTRE_TEAM_LEADER,
+                    ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_JUDGE, ST_CIC_SENIOR_JUDGE)
+                .grantHistoryOnly(
+                    ST_CIC_CASEWORKER,
+                    ST_CIC_SENIOR_CASEWORKER,
```
Comment (JakeCG) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerReferToLegalOfficer.java:66]: Do we need to remove duplication here?

Comment (JakeCG) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerReferToLegalOfficer.java:66]: in the grant and grantHistoryOnly

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCancelHearing.java [hunk 6]
```diff
@@ -70,28 +76,35 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
+                .description("Hearings: Cancel hearing")
+                .showSummary()
+                .aboutToStartCallback(this::aboutToStart)
+                .aboutToSubmitCallback(this::aboutToSubmit)
+                .submittedCallback(this::hearingCancelled)
+                .grant(CREATE_READ_UPDATE, SUPER_USER,
+                    ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN,
+                    ST_CIC_HEARING_CENTRE_TEAM_LEADER, ST_CIC_SENIOR_JUDGE)
+                .grantHistoryOnly(
+                    ST_CIC_CASEWORKER,
```
Comment (JakeCG) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCancelHearing.java:93]: Duplication here as well in the grant and grantHistoryOnly

**Review by pr00279 — DISMISSED** [review]

**Review by rajaram-nagarajan — DISMISSED** [review]

**Review by tomxelliott — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCancelHearing.java (hunk 6 above)
Comment (oleon23) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCancelHearing.java]: Thanks, have updated the permissions accordingly

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerReferToLegalOfficer.java (hunk 5 above)
Comment (oleon23) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerReferToLegalOfficer.java]: Thanks, have updated the permissions accordingly

**Review by tomxelliott — DISMISSED** [review]

**Pull request closed without merging**